src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private TruckProducer truckProducer = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = Collections.emptyList();
+        if (Bulldozer.class == type) {
+            machines = bulldozerProducer.get();
+        } else if (Excavator.class == type) {
+            machines = excavatorProducer.get();
+        } else if (Truck.class == type) {
+            machines = truckProducer.get();
+        }
+        return new ArrayList<>(machines);
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
```

Comment:
The `fill` method uses `List<? super Machine>` for the `machines` parameter, which is correct. However, consider checking if the list is not empty before attempting to fill it to avoid unnecessary operations.